libs/audit-service/src/main/java/uk/gov/di/ipv/core/library/auditing/extension/AuditExtensionAccountIntervention.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.di.ipv.core.library.auditing.extension;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+import uk.gov.di.ipv.core.library.annotations.ExcludeFromGeneratedCoverageReport;
+
+@ExcludeFromGeneratedCoverageReport
+@Getter
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class AuditExtensionAccountIntervention implements AuditExtensions {
+
+    private static final String REPROVE_IDENTITY_TYPE = "reprove_identity";
+
+    @JsonProperty private Boolean success;
+    @JsonProperty private final String type;
+
+    private AuditExtensionAccountIntervention(String type) {
+        this.type = type;
+    }
+
+    private AuditExtensionAccountIntervention(String type, Boolean success) {
+        this.type = type;
+        this.success = success;
+    }
```

Comment:
This would be fractionally more concise as a record class:
```
@ExcludeFromGeneratedCoverageReport
@JsonInclude(JsonInclude.Include.NON_NULL)
public record AuditExtensionAccountIntervention(String type, Boolean success) implements AuditExtensions {
    private static final String REPROVE_IDENTITY_TYPE = "reprove_identity";

    public static AuditExtensionAccountIntervention
            newReproveIdentityAuditExtensionAccountIntervention() {
        return new AuditExtensionAccountIntervention(REPROVE_IDENTITY_TYPE, null);
    }

    public static AuditExtensionAccountIntervention
            newReproveIdentityAuditExtensionAccountIntervention(boolean success) {
        return new AuditExtensionAccountIntervention(REPROVE_IDENTITY_TYPE, success);
    }
}
```
(Not a big deal)

Reply:
The only reason I did a class was to have a private constructor and enforce the use of the static functions to create the extension so that it controls what types can be created

Reply:
Ah of course, you can't hide the default constructor with a record 🤔 

TBH I don't think that's a massive deal (if you want to enforce safety, you could make the `type` an Enum, perhaps) - but happy for you to take either approach.